rollout/sync.go

```diff
@@ -691,6 +691,14 @@ func (c *rolloutContext) persistRolloutStatus(newStatus *v1alpha1.RolloutStatus)
 
 	prevStatus := c.rollout.Status
 	c.pauseContext.CalculatePauseStatus(newStatus)
+	if c.rollout.Spec.TemplateResolvedFromRef {
+		workloadRefObservation, _ := annotations.GetWorkloadGenerationAnnotation(c.rollout)
+		currentWorkloadObservedGeneration, _ := strconv.ParseInt(newStatus.WorkloadObservedGeneration, 10, 32)
+		if workloadRefObservation != int32(currentWorkloadObservedGeneration) {
+			newStatus.WorkloadObservedGeneration = strconv.Itoa(int(workloadRefObservation))
+		}
+	}
```

Comment:
To handle the case someone switches from workloadRef to inlined template, can we have an else case that does:

if c.rollout.Spec.TemplateResolvedFromRef {
 ...
} else {
  newStatus.WorkloadObservedGeneration = ""
}

Reply:
Added the else block. Thanks.